src/main/java/com/project/educationLab/domain/auth/service/PrincipalOauth2UserService.java

```diff
@@ -0,0 +1,67 @@
+package com.project.educationLab.domain.auth.service;
+
+import com.project.educationLab.domain.auth.dto.PrincipalDetails;
+import com.project.educationLab.domain.auth.dto.UserJoinRequest;
+import com.project.educationLab.domain.user.entity.User;
+import com.project.educationLab.domain.user.repository.UserRepository;
+import com.project.educationLab.domain.user.service.UserService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+@Service
+@Slf4j
+public class PrincipalOauth2UserService extends DefaultOAuth2UserService {
+
+    private final UserService userService;
+
+    public PrincipalOauth2UserService(UserService userService) {
+        this.userService = userService;
+    }
+
+    // "/login/oauth2/code/{providerId}" 요청시 자동으로 동작
+    // 소셜 로그인시 결과가 OAuth2User 타입 객체에 담김
+    @Override
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        log.info("getClientRegistration : {}", userRequest.getClientRegistration());
+        log.info("getAccessToken : {}", userRequest.getAccessToken().getTokenValue());
+        log.info("getAdditionalParameters : {}", userRequest.getAdditionalParameters());
+
+        OAuth2User oAuth2User = super.loadUser(userRequest);
+        log.info("getAttributes : {}", oAuth2User.getAttributes());
+        log.info("getAuthorities : {}", oAuth2User.getAuthorities());
+
+        String providerId = userRequest.getClientRegistration().getRegistrationId();
+        Map<String, Object> response = oAuth2User.getAttribute("response");
+        String userId = (String) response.get("id");
+        String username = providerId + "_" + userId;  // 중복 불허
```

Comment:
```suggestion
        String username = String.format("%s_%s",providerId, userId);  // 중복 불허
```
위와 같이 하거나 따로 함수로 빼는게 좋겠습니다.